modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java

```diff
@@ -4362,14 +4374,61 @@ private Map<String, String> _getClassNameIdStringUtilReplaceValues() {
 		return map;
 	}
 
-	private Serializable _getExpandoAttributeValue(Object object) {
-		if (object instanceof BigDecimal) {
-			BigDecimal bigDecimal = (BigDecimal)object;
+	private Serializable _getExpandoAttributeValue(JSONObject jsonObject) throws Exception{
 
-			return bigDecimal.doubleValue();
+		if(jsonObject.getInt("dataType") == ExpandoColumnConstants.BOOLEAN){
+			return jsonObject.getBoolean("defaultValue");
 		}
-
-		return (Serializable)object;
+		else if(jsonObject.getInt("dataType") == ExpandoColumnConstants.DATE){
+			DateFormat dateFormat = DateUtil.getISOFormat(jsonObject.getString("defaultValue"));
+			Date date = dateFormat.parse(jsonObject.getString("defaultValue"));
+			return date;
+		}
+		else 	if(jsonObject.getInt("dataType") == ExpandoColumnConstants.DOUBLE){
+			return jsonObject.getDouble("defaultValue");
+		}
+		else if(jsonObject.getInt("dataType") == ExpandoColumnConstants.DOUBLE_ARRAY){
+			return JSONUtil.toDoubleArray(jsonObject.getJSONArray("defaultValue"));
+		}
+		else if(jsonObject.getInt("dataType") == ExpandoColumnConstants.FLOAT){
+			return jsonObject.getDouble("defaultValue");
+		}
+		else if(jsonObject.getInt("dataType") == ExpandoColumnConstants.FLOAT_ARRAY){
+			return JSONUtil.toFloatArray(jsonObject.getJSONArray("defaultValue"));
+		}
+		else if (jsonObject.getInt("dataType") == ExpandoColumnConstants.GEOLOCATION) {
+			ExpandoValue value = new ExpandoValueImpl();
+			Geo geo = new Geo();
+			value.setGeolocationJSONObject(jsonObject.getJSONObject("defaultValue"));
+			geo.setLatitude(value.getGeolocationJSONObject().getDouble("latitude"));
+			geo.setLongitude(value.getGeolocationJSONObject().getDouble("longitude"));
```

Comment:
can't you use Geo.toDTO for this?

Reply:
yes, we can use Geo.toDTO for this, thank you for sharing with us, now I know a new way for to do this.

Reply:
Needs to do tests with this change, I did it there, but it doesn't work. 
Cast problem:

`java.lang.ClassCastException: com.liferay.headless.commerce.admin.catalog.dto.v1_0.Geo cannot be cast to com.liferay.portal.kernel.json.JSONObject`